src/SfxWeb/src/app/ViewModels/TreeNodeGroupViewModel.ts

```diff
@@ -303,52 +381,66 @@ export class TreeNodeGroupViewModel implements ITreeNode {
         }
     }
 
-    public select(actionDelay?: number, skipSelectAction?: boolean) {
-        if (this.tree.selectNode(this)) {
-            if (this.node.selectAction && !skipSelectAction) {
-                setTimeout(() => {
-                    if (this.selected) {
-                        this.node.selectAction();
-                    }
-                }, actionDelay || 0);
-            }
+    public select() : boolean {
+        return this.tree.selectNode(this);
+    }
+
+    public selectAndInteract() {
+        if(!this.selected){
+            this.select();
+        }
+        if (this.node.selectAction) {
+            this.node.selectAction();
+
         }
     }
 
-    public selectNext(actionDelay?: number) {
+    public selectNext() {
         if (this.hasExpandedAndLoadedChildren) {
-            this.displayedChildren[0].select(this.keyboardSelectActionDelayInMilliseconds);
+            this.displayedChildren[0].select();
         } else {
-            this.selectNextSibling();
+            this.moveToNextSibling()?.select();
         }
     }
 
-    public selectPrevious(actionDelay?: number) {
+    public selectPrevious() {
         const parentsChildren = this.getParentsChildren();
         const myIndex = parentsChildren.indexOf(this);
 
-        if (myIndex === 0 && this.parent) {
-            this.parent.select(this.keyboardSelectActionDelayInMilliseconds);
+        if (myIndex === 0 && this.parent && this.parent.nodeId !== 'base') {
+            this.parent.select();
         } else if (myIndex !== 0) {
             parentsChildren[myIndex - 1].selectLast();
         }
     }
 
+    private moveToRoot() {
+        return this.tree.childGroupViewModel.displayedChildren[0];
+    }
+
+    public selectRoot() {
+        this.moveToRoot().select();
+        
+    }
+
+    public selectEnd() {
+        this.moveToRoot().selectLast();
+    }
+
     public expandOrMoveToChild() {
-        if (this.hasChildren) {
-            if (this.isCollapsed) {
-                this.toggle();
-            } else {
-                this.selectNext();
-            }
+        if (this.isCollapsed) {
+            this.toggle();
+        }
+        else if(this.hasExpandedAndLoadedChildren) { //hasChildren is not set properly for leaf nodes, checking actual children length instead
```

Comment:
To work with expanders, hasChildren is true when children are not loaded. Expanders triggers the getChildren(), and set hasChildren properly. But they need to only exist for nodes with children, so for leaf nodes which don't have expanders, it remains to be true. I've added a check so in the rare cases where getChildren() is triggered for leaf nodes, hasChildren is set to false, but otherwise we should check for leaf nodes when possibly interacting with one.